### PR TITLE
removing ugly error message of #5147

### LINKED ISSUE #5147: Uncaught exception Term.DestKO ; Functional Scheme on nat
Note: the issue was created automatically with bugzilla2github tool

Original bug ID: BZ#5147
From: Jérémy Hervé &lt;<jeremy.herve@gmail.com>&gt;
Reported version: 8.5
CC:   @ejgallego, @forestjulien, @JasonGross, @ppedrot

### PR DESCRIPTION
**Pull request by forestjulien**
**Kind:** documentation / bug fix / feature / performance / infrastructure.
King:bugfix

Fixes / closes #5147

### CODE REVIEW

**Review by gares — CHANGES_REQUESTED** [review]
Clarify the error message

File: plugins/funind/functional_principles_types.ml [hunk 1]
```diff
@@ -628,16 +628,19 @@ let build_scheme fas =
...
                   (str "Cannot find " ++ Libnames.pr_reference f)
 	    in
-	    let evd',f = Evd.fresh_global (Global.env ()) !evd f_as_constant in
+            let evd',f = Evd.fresh_global (Global.env ()) !evd f_as_constant in
             let _ = evd := evd' in 
 	    let _ = Typing.e_type_of ~refresh:true (Global.env ()) evd (EConstr.of_constr f) in 
-	    (destConst f,sort)
+            if isConst f
+            then (destConst f,sort)
+            else user_err Pp.(pr_constr_env (Global.env ()) !evd f ++spc () ++ str "should be a function")
```
[the remark this answers is not in the document]
Comment (forestjulien) [reply plugins/funind/functional_principles_types.ml]: Done

**Review by gares — APPROVED** [review]

**Comment by gares** [comment]
Assigning to the other owner of this code.

**Comment by maximedenes** [comment]
@Matafou I believe the failure is unrelated, and was happening because of heavy load on Appveyor's VMs.

**Pull request merged**